### PR TITLE
Add MLOps Pipeline For Time Series Prediction: Binance Trading article

### PR DESCRIPTION
**Pull request by BexTuychiev**
Added and updated the GitHub repository for the following article: https://neptune.ai/blog/mlops-pipeline-for-time-series-prediction-tutorial

### CODE REVIEW

File: community-code/binance-trading-neptune-master/notebooks/experiment.ipynb [hunk 1]
```diff
@@ -0,0 +1,1414 @@
...
+     "output_type": "stream",
+     "text": [
+      "\u001b[32m[I 2022-06-20 17:31:10,770]\u001b[0m Trial 19 finished with value: 4.373678616655161 and parameters: {'n_estimators': 832, 'max_depth': 6, 'learning_rate': 0.05645751761800897, 'subsample': 0.8949849334127633, 'colsample_bytree': 0.8418850562727114, 'gamma': 17, 'look_back': 70}. Best is trial 13 with value: 5.210315443125857.\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "import optuna\n",
+    "import warnings\n",
+    "import neptune.new as neptune\n",
```
Comment (SiddhantSadangi) [inline community-code/binance-trading-neptune-master/notebooks/experiment.ipynb:1052]: @BexTuychiev / @domikoz - this is the legacy `neptune-client` API: https://docs.neptune.ai/setup/upgrading/

Are we still using this in blogs?

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
This would need refactoring to use the new Neptune >1.0 API

File: community-code/README.md [hunk 2]
```diff
@@ -5,12 +5,19 @@
...
-    </ul>
-  <br>
-  <h4> Notice something wrong? </h4>
-  Drop us a mail at <a href="mailto:marketing@neptune.ai"> marketing@neptune.ai</a>.
+</div>
+<!-- Add in alphabetical order -->
+
+| Title | Blog | Code | Neptune
+| --- | :---: | :---: | :---:
+| MLOps For Time Series Prediction: Binance Trading Tutorial | [![blog]](https://neptune.ai/blog/mlops-pipeline-for-time-series-prediction-tutorial) | [![github]](community-code/binance-trading-neptune-master) | [![neptune]](https://app.neptune.ai/enes.zvornicanin/optuna-test/runs/table?viewId=standard-view&detailsTab=metadata&dash=table&type=run)
```
[the remark this answers is not in the document]
Comment (domikoz) [reply community-code/README.md:13]: We should have it in the community workspace, the project space is ready; @BexTuychiev could you please take care of that? Thanks!

File: community-code/binance-trading-neptune-master/notebooks/experiment.ipynb (hunk 1 above)
Comment (domikoz) [reply community-code/binance-trading-neptune-master/notebooks/experiment.ipynb:1052]: @BexTuychiev please make sure everything is up-to-date 🙏

**Comment by domikoz** [comment]
> This would need refactoring to use the new Neptune >1.0 API

@BexTuychiev

File: community-code/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (BexTuychiev) [reply community-code/README.md:13]: I am working on recreating the project in the community space

File: community-code/binance-trading-neptune-master/notebooks/experiment.ipynb (hunk 1 above)
Comment (BexTuychiev) [reply community-code/binance-trading-neptune-master/notebooks/experiment.ipynb]: I made another commit changing all instances of outdated code. Also, updated the dependencies

File: community-code/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (BexTuychiev) [reply community-code/README.md]: @SiddhantSadangi 

I added a link to the community project.

File: community-code/binance-trading-neptune-master/notebooks/experiment.ipynb [hunk 3]
```diff
@@ -0,0 +1,1018 @@
...
+    "\n",
+    "hb = HourlyBacktester(dt.data, trader_config)\n",
+    "n_trials = [15, 20, 25, 30]\n",
+    "n_studies = 4\n",
+    "\n",
+    "for i in range(n_studies):\n",
+    "    run = neptune.init_run(\n",
+    "        project=os.getenv(\"NEPTUNE_PROJECT_NAME\"),\n",
+    "        api_token=os.getenv(\"NEPTUNE_API_TOKEN\"),\n",
+    "    )\n",
```
Comment (SiddhantSadangi) [inline community-code/binance-trading-neptune-master/notebooks/experiment.ipynb:908]: Neptune will pick ENV variables automatically. No need to use `os.getenv()`.

```suggestion
    "    run = neptune.init_run()\n",
```

File: community-code/binance-trading-neptune-master/out/placeholder.txt
Comment (SiddhantSadangi) [inline community-code/binance-trading-neptune-master/out/placeholder.txt:1]: Is this added just to include the `out/` folder in git?
If yes, prefer to use an empty `.gitkeep` file as per convention.

File: community-code/binance-trading-neptune-master/src/paper_trading.py [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+from datetime import datetime, timedelta
+import pandas as pd
+import numpy as np
+from binance.client import Client
+from xgboost import XGBClassifier
+import talib
+import optuna
+from tqdm import tqdm
+from unittest.mock import Mock
+import neptune as neptune
```
Comment (SiddhantSadangi) [inline community-code/binance-trading-neptune-master/src/paper_trading.py:12]:
```suggestion
import neptune
```

File: community-code/binance-trading-neptune-master/src/paper_trading.py [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+        df = df_his.append(df_curr)
+        df = df.drop_duplicates(subset=["date"])
+
+        self.df_training_data = self.generate_features(df)
+        self.df_training_data = self.df_training_data.dropna()
+
+        run = neptune.init_run(
+            project=os.getenv("NEPTUNE_PROJECT_NAME"),
+            api_token=os.getenv("NEPTUNE_API_TOKEN"),
+        )
```
Comment (SiddhantSadangi) [inline community-code/binance-trading-neptune-master/src/paper_trading.py:266]: Same as earlier

```suggestion
        run = neptune.init_run()
```

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]

File: community-code/binance-trading-neptune-master/out/placeholder.txt
Comment (BexTuychiev) [reply community-code/binance-trading-neptune-master/out/placeholder.txt:1]: Yes, that's correct. I removed the placeholder file and added an empty `.gitkeep` in a new commit. I didn't know about this convention, thanks for teaching!

File: 0001-Add-.gitkeep-file-to-preserve-out-folder-in-the-dire.patch
Comment (SiddhantSadangi) [inline 0001-Add-.gitkeep-file-to-preserve-out-folder-in-the-dire.patch:1]: Can you remove this patch file from the commit?

Comment (BexTuychiev) [reply 0001-Add-.gitkeep-file-to-preserve-out-folder-in-the-dire.patch:1]: Just removed it, can you please check?

**Review by SiddhantSadangi — APPROVED** [review]

**Pull request merged**